packages/eas-cli/src/commands/env/create.ts

```diff
@@ -0,0 +1,251 @@
+            const confirmation = await confirmAsync({
+              message: `Shared variable ${name} already exists on this project. Do you want to unlink it first?`,
+            });
+
+            if (!confirmation) {
+              Log.log('Aborting');
+              return;
+            }
+          } else if (!force) {
+            throw new Error(
+              `Shared variable ${name} already exists on this project. Use --force to overwrite it.`
+            );
+          }
+
+          await EnvironmentVariableMutation.unlinkSharedEnvironmentVariableAsync(
+            graphqlClient,
+            existingVariable.id,
+            projectId,
+            environment
+          );
+          Log.withTick(
+            `Unlinking shared variable ${chalk.bold(name)} on project ${chalk.bold(
+              projectDisplayName
+            )}.`
+          );
+        } else {
+          if (!nonInteractive) {
+            const confirmation = await confirmAsync({
+              message: `Variable ${name} already exists on this project. Do you want to overwrite it?`,
+            });
+
+            if (!confirmation) {
+              Log.log('Aborting');
+              return;
+            }
+          } else if (!force) {
+            throw new Error(
+              `Shared variable ${name} already exists on this project. Use --force to overwrite it.`
+            );
+          }
+          overwrite = true;
+        }
+      }
+
+      const variable = await EnvironmentVariableMutation.createForAppAsync(
+        graphqlClient,
+        {
+          name,
+          value,
+          environment,
+          visibility,
+          overwrite,
+        },
+        projectId
+      );
+      if (!variable) {
+        throw new Error(
+          `Could not create variable with name ${name} on project ${projectDisplayName}`
+        );
+      }
+
+      Log.withTick(
+        `Created a new variable ${chalk.bold(name)} on project ${chalk.bold(projectDisplayName)}.`
+      );
+    } else if (scope === EnvironmentVariableScope.Shared) {
+      const sharedVariables = await EnvironmentVariablesQuery.sharedAsync(graphqlClient, projectId);
+      const existingVariable = sharedVariables.find(variable => variable.name === name);
+      if (existingVariable) {
+        throw new Error(
+          `Shared variable with ${name} name already exists on this account. Use a different name or delete the existing variable.`
+        );
+      }
+
+      if (environment && !link) {
+        if (!nonInteractive) {
+          const confirmation = await confirmAsync({
+            message: `Unexpected argument: --environment can only be used with --link flag. Do you want to link the variable to the current project?`,
+          });
+
+          if (!confirmation) {
+            Log.log('Aborting');
+            return;
+          }
+        } else {
+          throw new Error('Unexpected argument: --environment can only be used with --link flag.');
+        }
```

Comment:
I believe that this could also be moved to the `validateFlags`/`sanitizeFlags` function, but this is more or less a nit